torchgeo/datasets/spacenet.py

```diff
@@ -191,49 +199,287 @@ def __getitem__(self, index: int) -> Dict[str, Tensor]:
         """
-        if self._check_integrity():
-            print("Files already downloaded")
-            return
-
-        download_radiant_mlhub_dataset(self.dataset_id, self.root, api_key)
-        archive_path = os.path.join(self.root, self.foldername + ".tar.gz")
-        if (
-            self.checksum
-            and check_integrity(archive_path, self.md5)
-            or not self.checksum
-        ):
+        for collection in collections:
+            download_radiant_mlhub_collection(collection, self.root, api_key)
+            archive_path = os.path.join(self.root, collection + ".tar.gz")
+            if (
+                not self.checksum
+                or not check_integrity(
+                    archive_path, self.collection_md5_dict[collection]
+                )
+            ) and self.checksum:
+                raise RuntimeError(f"Collection {collection} corrupted")
+
+            print("Extracting...")
             extract_archive(archive_path)
-        else:
-            raise RuntimeError("Dataset corrupted")
+
+
+class SpaceNet1(SpaceNet):
+    """SpaceNet 1: Building Detection v1 Dataset.
+
+    `SpaceNet 1 <https://spacenet.ai/spacenet-buildings-dataset-v1/>`_
+    is a dataset of building footprints over the city of Rio de Janeiro.
+
+    * *Dataset features*:
+
+        * No. of images: 6940 (8 Band) + 6940 (RGB)
+        * No. of polygons: 382,534 building labels
+        * Area Coverage: 2544 sq km
+        * GSD: 1 m (8 band),  50 cm (rgb)
+        * Chip size: 102 x 110 (8 band), 407 x 439 (rgb)
+
+    .. note::
+       Chip size of both imagery can have 1 pixel difference
+
+    * *Dataset format*:
+
+        * Imagery - Worldview-2 GeoTIFFs
+                * 8Band.tif (Multispectral)
+                * RGB.tif (Pansharpened RGB)
+        * Labels - GeoJSON
+                * labels.geojson
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/1807.01232
+
+    .. note::
+
+       This dataset requires the following additional library to be installed:
+
+       * `radiant-mlhub <https://pypi.org/project/radiant-mlhub/>`_ to download the
+         imagery and labels from the Radiant Earth MLHub
+
+    """
+
+    dataset_id = "spacenet1"
+    imagery = {"rgb": "RGB.tif", "8band": "8Band.tif"}
+    label_glob = "labels.geojson"
+    collection_md5_dict = {"sn1_AOI_1_RIO": "e6ea35331636fa0c036c04b3d1cbf226"}
+
+    def __init__(
+        self,
+        root: str,
+        image: str = "rgb",
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        download: bool = False,
+        api_key: Optional[str] = None,
+        checksum: bool = False,
+    ) -> None:
+        """Initialise a new SpaceNet 1 Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            image: image selection which must be "rgb" or "8band"
```

Comment:
Do we have an assert statement anywhere to enforce that `image` is valid?

Reply:
We do now